python/cuml/dask/common/input_utils.py

```diff
@@ -0,0 +1,314 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import cudf
+import cupy as cp
+import dask.array as da
+
+from cuml.dask.common.dask_df_utils import to_dask_cudf
+from cuml.utils.memory_utils import with_cupy_rmm
+
+from collections import OrderedDict
+from cudf.core import DataFrame
+from dask_cudf.core import DataFrame as dcDataFrame
+
+from dask import delayed
+from dask.distributed import wait
+from dask.distributed import default_client
+from tornado import gen
+from toolz import first
+from collections.abc import Iterable
+from dask.array.core import Array as daskArray
+from uuid import uuid1
+
+from collections import defaultdict
+from functools import reduce
+
+import dask.dataframe as dd
+
+
+class MGData:
+
+    def __init__(self, gpu_futures=None, worker_to_parts=None, workers=None,
+                 datatype=None, multiple=False, client=None):
+        self.client = default_client() if client is None else client
+        self.gpu_futures = gpu_futures
+        self.worker_to_parts = worker_to_parts
+        self.workers = workers
+        self.datatype = datatype
+        self.multiple = multiple
+        self.worker_info = None
+        self.total_rows = None
+        self.ranks = None
+        self.parts_to_sizes = None
+        self.total_rows = None
+
+    """ Class methods for initalization """
+
+    @classmethod
+    def single(cls, data, client=None):
+        gpu_futures = client.sync(_extract_partitions, data, client)
+
+        worker_to_parts = _workers_to_parts(gpu_futures)
+
+        workers = list(map(lambda x: x[0], worker_to_parts.items()))
+
+        datatype = 'cudf' if isinstance(data, dcDataFrame) else 'cupy'
+
+        return MGData(gpu_futures=gpu_futures, worker_to_parts=worker_to_parts,
+                      workers=workers, datatype=datatype, multiple=False,
+                      client=client)
+
+    @classmethod
+    def colocated(cls, data, force=False, client=None):
```

Comment:
Is force actually used?

Reply:
Not right now. In fact I'm not sure it is needed: if data is not colocated then the zipping into tuple will do it, but if its already colocated the zipping won't affect it, so we can just safely remove the whole `force_colocality` argument, no? And of course this `force` as well